### PR TITLE
GEOPY-1348

### PR DESCRIPTION
**Pull request by domfournier**

### CODE REVIEW

File: curve_apps/trend_lines/driver.py [hunk 1]
```diff
@@ -108,12 +108,14 @@ def get_connections(self) -> tuple:
             if len(ind) < 2:
                 continue
 
+            detection_params = dict(self.params.detection)
+            if detection_params["max_distance"] is None:
+                detection_params["max_distance"] = np.inf
```
Comment (benk-mira) [inline curve_apps/trend_lines/driver.py:113]: To go along with my comment on geoapps-utils, could this work?:
```
segments = find_curves(
    self.vertices[ind, :2],
    self.parts[ind],
    self.params.detection
)
```
You'd obviously have to move some of your logic over to the opts class..

File: curve_apps/trend_lines/params.py [hunk 2]
```diff
@@ -83,11 +83,15 @@ class DetectionParameters(BaseModel):
...
+    :param azimuth: Azimuth of the path.
+    :param azimuth_tol: Tolerance for the azimuth of the path.
+    :param damping: Damping factor between [0, 1] for the path roughness.
     :param min_edges: Minimum number of points in a curve.
     :param max_distance: Maximum distance between points in a curve.
-    :param damping: Damping factor between [0, 1] for the path roughness.
     """
 
+    azimuth: Optional[float] = None
+    azimuth_tol: Optional[float] = None
```
Comment (benk-mira) [inline curve_apps/trend_lines/params.py:94]: Hey... When I try to use Optional pre-commit converts to float | None (think it's the pyupgrade hook).  Did you have to do something to get around it?

**Review by benk-mira — COMMENTED** [review]
All good besides the recommendation to move detection params definition to geoapps-utils next to the function that uses the parameters

File: curve_apps/trend_lines/driver.py (hunk 1 above)
Comment (domfournier) [reply curve_apps/trend_lines/driver.py:113]: yep, moved it over.

File: curve_apps/trend_lines/params.py (hunk 2 above)
Comment (domfournier) [reply curve_apps/trend_lines/params.py:94]: Yep, as discussed you need to exclude the file in the pre-commit-config

**Review by benk-mira — APPROVED** [review]
Thanks for the edit

**Review by benk-mira — APPROVED** [review]
Not sure I follow all the pydantic details.. If it passes tests, great!

**Pull request merged**